nautobot/ipam/tests/test_filters.py

```diff
@@ -866,6 +869,88 @@ def test_role(self):
+        devicestatus = Status.objects.get_for_model(Device).first()
+        device = Device.objects.create(
+            name="Device 1",
+            location=location,
+            device_type=devicetype,
+            role=devicerole,
+            status=devicestatus,
+        )
+        int_status = Status.objects.get_for_model(Interface).first()
+        int_type = InterfaceTypeChoices.TYPE_1GE_FIXED
+        interfaces = [
+            Interface.objects.create(device=device, name="eth0", status=int_status, type=int_type),
+            Interface.objects.create(device=device, name="eth1", status=int_status, type=int_type),
+            Interface.objects.create(device=device, name="eth2", status=int_status, type=int_type),
+        ]
+
+        clustertype = ClusterType.objects.create(name="Cluster Type 1")
+        cluster = Cluster.objects.create(cluster_type=clustertype, name="Cluster 1")
+        vm_status = Status.objects.get_for_model(VirtualMachine).first()
+        virtual_machine = (VirtualMachine.objects.create(name="Virtual Machine 1", cluster=cluster, status=vm_status),)
+        vm_int_status = Status.objects.get_for_model(VMInterface).first()
+        vm_interfaces = [
+            VMInterface.objects.create(virtual_machine=virtual_machine[0], name="veth0", status=vm_int_status),
+            VMInterface.objects.create(virtual_machine=virtual_machine[0], name="veth1", status=vm_int_status),
+        ]
+
+        IPAddressToInterface.objects.create(ip_address=ip_addresses[0], interface=interfaces[0], vm_interface=None)
+        IPAddressToInterface.objects.create(ip_address=ip_addresses[1], interface=interfaces[1], vm_interface=None)
+        IPAddressToInterface.objects.create(ip_address=ip_addresses[2], interface=interfaces[2], vm_interface=None)
+        IPAddressToInterface.objects.create(ip_address=ip_addresses[3], interface=None, vm_interface=vm_interfaces[0])
+        IPAddressToInterface.objects.create(ip_address=ip_addresses[4], interface=None, vm_interface=vm_interfaces[1])
+
+    def test_boolean_filters(self):
+        filters = [
+            "is_source",
+            "is_destination",
+            "is_default",
+            "is_preferred",
+            "is_primary",
+            "is_secondary",
+            "is_standby",
+        ]
+
+        ip_association1 = IPAddressToInterface.objects.first()
+        ip_association2 = IPAddressToInterface.objects.last()
+        for test_filter in filters:
+            setattr(ip_association1, test_filter, True)
+            setattr(ip_association2, test_filter, True)
+        ip_association1.validated_save()
+        ip_association2.validated_save()
+
+        for test_filter in filters:
+            with self.subTest(filter=test_filter):
+                params = {test_filter: True}
+                self.assertQuerysetEqualAndNotEmpty(
+                    self.filterset(params, self.queryset).qs, self.queryset.filter(**{test_filter: True}), ordered=False
+                )
```

Comment:
Should we instead do something like 
```suggestion
                self.assertQuerysetEqualAndNotEmpty(
                    self.filterset(params, self.queryset).qs.order_by('pk'), self.queryset.filter(**{test_filter: True}).order_by('pk')
                )
```

Reply:
The intent of this test is to test the filterset return values. Since this model isn't ordered we can't expect any kind of order on the ORM or API. From [the docs](https://docs.djangoproject.com/en/4.2/ref/models/options/#ordering):

> If a query doesn’t have an ordering specified, results are returned from the database in an unspecified order.